index.js

```diff
@@ -119,6 +119,13 @@ Router.prototype._on = function _on (method, path, opts, handler, store) {
     // search for parametric or wildcard routes
     // parametric route
     if (path.charCodeAt(i) === 58) {
+      if (i !== len - 1 && path.charCodeAt(i + 1) === 58) {
```

Comment:
I'm wondering if we can check if the preceding char is a slash (47) and if not avoid creating the parametric node.

eg:
```
/foo/:bar => char before : is / => parametric route
/foo/bar:baz => char before : is not / => static route
```

Reply:
I was thinking about that. But what if I want a parameter following some static string?
Maybe something like `/tests/issue-:num(^\d+$)/description` to route to `/tests/issue-175/description`

Reply:
The same thing, we can check for the preceding char code (45 in this case) and act accordingly.
I feel that having two columns is counterintuitive and obscure. I prefer to have a transparent API and the doc explains how to achieve the specific use cases like the one you've mentioined.

Reply:
So which syntax do you think is more intuitive?
I agree that two colons is not very intuitive. 
But I was thinking a lot of people came to fastify from express.js and Koa. (https://github.com/expressjs/express/issues/3857)
I could replace two colons with backslash for example

Reply:
I think the more intuitive is the one I've described in my first comment.
I agree that familiarity with express/koa is nice to have, but not when we disagree with a pattern.
@mcollina what do you think?

Reply:
A change that would not require an additional `charCodeAt()` in the hot path would be better. However, this is not causing any regression so maybe this is also ok.

Reply:
@delvedor according to the current docs it's possible to have any number of parameters within the same couple of slash. And there are no exact delimiters (slash or something).
Example from the readme:
`router.on('GET', '/example/at/:hour(^\\d{2})h:minute(^\\d{2})m', (req, res, params) => {}))`
In this example the char before `:` is `h`. So based on the structure of the route, we cannot know for sure if I want to declare a new parameter or just use a colon.

Reply:
Great point @morigs!